src/solver/ts-generator/thermal.cpp

```diff
@@ -596,11 +597,80 @@ void GeneratorTempData::operator()(Data::Area& area, Data::ThermalCluster& clust
 }
 } // namespace
 
+
+std::vector<std::pair<std::string, std::string>> splitStringIntoPairs (const std::string& s,
+                                                                       char delimiter1,
+                                                                       char delimiter2)
+{
+    std::vector<std::pair<std::string, std::string>> pairs;
+    std::stringstream ss(s);
+    std::string token;
+
+    while (std::getline(ss, token, delimiter1))
```

Comment:
You may iterate on a range to simplify
```cpp
for (auto token: std::views::split(s, delimiter1) {
    ...
}
```

Reply:
token is not a string in this context and needs to be transformed in order to be used, increasing the complexity